pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -265,24 +265,44 @@ func (r ReplicaSetReconciler) ensureAutomationConfig(mdb mdbv1.MongoDB) error {
 	if err != nil {
 		return err
 	}
+
 	if err := r.client.CreateOrUpdate(&cm); err != nil {
 		return err
 	}
 	return nil
 }
 
-func buildAutomationConfig(mdb mdbv1.MongoDB, mdbVersionConfig automationconfig.MongoDbVersionConfig) automationconfig.AutomationConfig {
+func buildAutomationConfig(mdb mdbv1.MongoDB, mdbVersionConfig automationconfig.MongoDbVersionConfig, currentAc automationconfig.AutomationConfig) (automationconfig.AutomationConfig, error) {
 	domain := getDomain(mdb.ServiceName(), mdb.Namespace, "")
-	return automationconfig.NewBuilder().
+	newAc := automationconfig.NewBuilder().
 		SetTopology(automationconfig.ReplicaSetTopology).
 		SetName(mdb.Name).
 		SetDomain(domain).
 		SetMembers(mdb.Spec.Members).
 		SetMongoDBVersion(mdb.Spec.Version).
-		SetAutomationConfigVersion(1). // TODO: Correctly set the version
+		SetAutomationConfigVersion(currentAc.Version).
 		SetFCV(mdb.GetFCV()).
 		AddVersion(mdbVersionConfig).
 		Build()
+
+	/* Here we compare the bytes of the two automationconfigs,
+	 we can't use reflect.DeepEqual() as it treats nil entries as different from empty ones,
+	and in the AutomationConfig Struct we use omitempty to set empty field to nil */
+	newAcBytes, err1 := json.Marshal(newAc)
+	if err1 != nil {
```

Comment:
generally, we name all error values `err`. Even if there is name shadowing. If possible, we like to reduce the scope of the variable using the construct

```golang
if err := SomeFunc(); err != nil {
   // err is only in scope here
}
```

but this is not always possible with multi value return functions. I would prefer just re-using the same error variable and re-assigning rather than multiple similarly named error variables